### PR TITLE
Specialize ToString for Symbol

### PR DESCRIPTION
**Pull request by camsteffen**

### CODE REVIEW

**Comment by camsteffen** [comment]
A simple local bench of `sym::Display.to_string()` shows a consistent change from 80ns/iter to 71ns/iter.

File: compiler/rustc_span/src/symbol.rs [hunk 1]
```diff
@@ -1901,6 +1901,13 @@ impl fmt::Display for Symbol {
     }
 }
 
+// takes advantage of `str::to_string` specialization
+impl ToString for Symbol {
+    fn to_string(&self) -> String {
+        return self.as_str().to_string();
```
Comment (compiler-errors) [inline compiler/rustc_span/src/symbol.rs:1907]:
```suggestion
        self.as_str().to_string()
```

**Comment by compiler-errors** [comment]
@bors try @rust-timer queue

(though likely to show no results, since I guess heavy usage of `Symbol::to_string` happens on diagnostics paths only)

File: compiler/rustc_span/src/symbol.rs (hunk 1 above)
Comment (camsteffen) [reply compiler/rustc_span/src/symbol.rs:1907]: I need to take a mental break after switching from Python.

**Comment by compiler-errors** [comment]
Hm, maybe this PR is not worth the added impl? Unless @camsteffen has something in mind that benefits from this speed boost.

**Comment by camsteffen** [comment]
I could go either way. On one hand Symbol is very widely used and this is a low-cost common-sense optimization. On the other hand this probably won't make anyone's life better.

**Comment by compiler-errors** [comment]
I guess it doesn't hurt anything.

r? @compiler-errors @bors r+

**Comment by camsteffen** [comment]
@bors r=compiler-errors

Removed the needless `return`

**Comment by compiler-errors** [comment]
Thanks, forgot that hadn't gotten applied.

**Pull request merged**